File: app/src/main/java/com/minerdev/greformanager/OneSideDrawerLayout.java
package com.minerdev.greformanager;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.drawerlayout.widget.DrawerLayout;

public class OneSideDrawerLayout extends DrawerLayout {
    private Boolean isSwipeOpenEnabled = true;
    private int drawerGravity;

    public OneSideDrawerLayout(@NonNull Context context) {
        super(context);
    }

    public OneSideDrawerLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public OneSideDrawerLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (!isSwipeOpenEnabled && !isDrawerVisible(drawerGravity)) {
            return false;
        }

        return super.onInterceptTouchEvent(ev);
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        if (!isSwipeOpenEnabled && !isDrawerVisible(drawerGravity)) {
            return false;
        }

        return super.onTouchEvent(ev);
    }

    public void lockSwipe(int drawerGravity) {
        isSwipeOpenEnabled = false;
        this.drawerGravity = drawerGravity;
    }

    public void unLockSwipe() {
        isSwipeOpenEnabled = true;
    }
}

File: app/src/main/java/com/minerdev/greformanager/customview/OneSideDrawerLayout.kt
package com.minerdev.greformanager.customview

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.drawerlayout.widget.DrawerLayout

class OneSideDrawerLayout : DrawerLayout {
    private var isSwipeOpenEnabled = true
    private var drawerGravity = 0

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle)

    override fun onInterceptTouchEvent(ev: MotionEvent?): Boolean {
        return if (!isSwipeOpenEnabled && !isDrawerVisible(drawerGravity)) {
            false
        } else
            super.onInterceptTouchEvent(ev)
    }

    override fun onTouchEvent(ev: MotionEvent?): Boolean {
        return if (!isSwipeOpenEnabled && !isDrawerVisible(drawerGravity)) {
            false
        } else
            super.onTouchEvent(ev)
    }

    fun lockSwipe(drawerGravity: Int) {
        isSwipeOpenEnabled = false
        this.drawerGravity = drawerGravity
    }

    fun unLockSwipe() {
        isSwipeOpenEnabled = true
    }
}